src/bridge/ERC20Bridge.sol

```diff
@@ -50,20 +50,30 @@ contract ERC20Bridge is AbsBridge, IERC20Bridge {
         uint256 value,
         bytes memory data
     ) internal override returns (bool success, bytes memory returnData) {
+        address _nativeToken = nativeToken;
+
         // we don't allow outgoing calls to native token contract because it could
         // result in loss of native tokens which are escrowed by ERC20Bridge
-        if (to == nativeToken) {
-            revert CallTargetNotAllowed(nativeToken);
+        if (to == _nativeToken) {
+            revert CallTargetNotAllowed(_nativeToken);
         }
 
         // first release native token
-        IERC20(nativeToken).safeTransfer(to, value);
+        IERC20(_nativeToken).safeTransfer(to, value);
         success = true;
 
-        // if there's data do additional contract call
+        // if there's data do additional contract call. Make sure that call is not used to
+        // change bridge contract's balance of the native token
         if (data.length > 0) {
+            uint256 bridgeBalanceBefore = IERC20(_nativeToken).balanceOf(address(this));
+
             // solhint-disable-next-line avoid-low-level-calls
             (success, returnData) = to.call(data);
+
+            uint256 bridgeBalanceAfter = IERC20(_nativeToken).balanceOf(address(this));
+            if (bridgeBalanceAfter != bridgeBalanceBefore) {
```

Comment:
can be a bit looser? idk
```suggestion
            if (bridgeBalanceAfter < bridgeBalanceBefore) {
```

Reply:
yeah I don't see practical use case to increase balance, but it doesn't hurt. updated